test/blackbox-tests/test-cases/odoc/warn-error/run.t

```diff
@@ -0,0 +1,42 @@
+  >     -e '\#^.*/ocamlc.opt -config > /tmp/dune.*\.output$#d' \
+  >     -e "s#$(opam var prefix)#%OPAM%#"
+  > EOF
+
+All calls to `odoc compile` and `odoc html` should have the `--warn-error` option,
+except for the package `bar` (`page-index.odoc` is OK).
+
+  $ . ./dune_verbose.sh build @doc
+  (cd _build/default && %OPAM%/bin/ocamldep.opt -modules -intf sub_env/lib/bar.mli) > _build/default/sub_env/lib/.bar.objs/bar.mli.d
+  (cd _build/default && %OPAM%/bin/ocamlc.opt -w @1..3@5..28@30..39@43@46..47@49..57@61..62-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -bin-annot -I sub_env/lib/.bar.objs/byte -no-alias-deps -opaque -o sub_env/lib/.bar.objs/byte/bar.cmi -c -intf sub_env/lib/bar.mli)
+  (cd _build/default/_doc/_odoc/pkg/bar && %OPAM%/bin/odoc compile --pkg bar -o page-bar.odoc ../../../../sub_env/doc/bar.mld)
+  (cd _build/default/_doc/_odoc/pkg/bar && %OPAM%/bin/odoc compile --warn-error --pkg bar -o page-index.odoc ../../../_mlds/bar/index.mld)
+  (cd _build/default/sub_env/lib/.bar.objs/byte && %OPAM%/bin/odoc compile -I . -I ../../../../_doc/_odoc/pkg/bar --pkg bar -o bar.odoc bar.cmti)
+  (cd _build/default/_doc/_html && %OPAM%/bin/odoc html --warn-error -I ../_odoc/pkg/bar -I ../../sub_env/lib/.bar.objs/byte -o . ../_odoc/pkg/bar/page-bar.odoc)
+  (cd _build/default && %OPAM%/bin/odoc support-files -o _doc/_html)
+  (cd _build/default && %OPAM%/bin/ocamldep.opt -modules -intf lib/foo.mli) > _build/default/lib/.foo.objs/foo.mli.d
+  (cd _build/default && %OPAM%/bin/ocamlc.opt -w @1..3@5..28@30..39@43@46..47@49..57@61..62-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -bin-annot -I lib/.foo.objs/byte -no-alias-deps -opaque -o lib/.foo.objs/byte/foo.cmi -c -intf lib/foo.mli)
+  (cd _build/default/_doc/_odoc/pkg/foo && %OPAM%/bin/odoc compile --warn-error --pkg foo -o page-foo.odoc ../../../../doc/foo.mld)
+  (cd _build/default/_doc/_odoc/pkg/foo && %OPAM%/bin/odoc compile --warn-error --pkg foo -o page-index.odoc ../../../_mlds/foo/index.mld)
+  (cd _build/default/lib/.foo.objs/byte && %OPAM%/bin/odoc compile --warn-error -I . -I ../../../_doc/_odoc/pkg/foo --pkg foo -o foo.odoc foo.cmti)
+  (cd _build/default/_doc/_html && %OPAM%/bin/odoc html --warn-error -I ../_odoc/pkg/foo -I ../../lib/.foo.objs/byte -o . ../_odoc/pkg/foo/page-foo.odoc)
+  (cd _build/default/_doc/_html && %OPAM%/bin/odoc html --warn-error -I ../_odoc/pkg/bar -I ../../sub_env/lib/.bar.objs/byte -o . ../_odoc/pkg/bar/page-index.odoc)
+  (cd _build/default/_doc/_html && %OPAM%/bin/odoc html -I ../_odoc/pkg/bar -I ../../sub_env/lib/.bar.objs/byte -o . ../../sub_env/lib/.bar.objs/byte/bar.odoc)
+  (cd _build/default/_doc/_html && %OPAM%/bin/odoc html --warn-error -I ../_odoc/pkg/foo -I ../../lib/.foo.objs/byte -o . ../_odoc/pkg/foo/page-index.odoc)
+  (cd _build/default/_doc/_html && %OPAM%/bin/odoc html --warn-error -I ../_odoc/pkg/foo -I ../../lib/.foo.objs/byte -o . ../../lib/.foo.objs/byte/foo.odoc)
```

Comment:
This output feels a bit too verbose. In order to convince ourselves that the test is passing, we have to go through it to check that `--warn-error` is correctly passed where it should. What about actually leaving an odoc error and checking that it is reported as an error?

Reply:
Generating a warning from the `odoc html` is not easy and Odoc is called from multiple independent paths in the source code and I wasn't convinced each paths had the correct environment.
Would it be better if I do both ?

Reply:
Well, by testing with an error in the source file, we are really testing the user-facing aspect of this PR, which is the most important.

If it's difficult to do that, we can keep the current way, but the output should be cleaned up more. Basically, a new reader should be able to look at the test and quickly understand what it is doing and whether it is passing or not. Otherwise, if some day this test breaks we might just accept the new output without realising the test is no longer passing.

Reply:
I changed the tests to show the actual output. I agree it looks a bit better.